app/lib/common/utilities/genome_data.dart

```diff
@@ -0,0 +1,80 @@
+import 'dart:collection';
+import 'dart:convert';
+
+import 'package:http/http.dart';
+
+import '../../profile/models/hive/alleles.dart';
+import '../../profile/models/hive/cpic_lookup_response.dart';
+import '../../profile/models/hive/diplotype.dart';
+import '../constants.dart';
+import '../services.dart';
+
+Future<void> fetchAndSaveAllesData(String token, String url) async {
+  final userAlleleData = getBox<Alleles>(Boxes.alleles);
+  if (userAlleleData.get('alleles') == null) {
+    final response = await getStarAlleles(token, url);
+    if (response.statusCode == 200) {
+      await _saveAlleleData(response);
+    } else {
+      throw Exception('Error occurred during loading of allele data');
+    }
+  }
+}
+
+Future<Response> getStarAlleles(String? token, String url) async =>
+    get(Uri.parse(url), headers: {'Authorization': 'Bearer $token'});
+
+Future<void> _saveAlleleData(Response response) async {
+  final json = jsonDecode(response.body);
+  final alleles = Alleles.fromJson(json);
+  alleles.diplotypes = alleles.diplotypes.filterValidDiplotypes();
+  return getBox<Alleles>(Boxes.alleles).put('alleles', alleles);
+}
+
+Future<void> fetchAndSaveLookups() async {
+  if (!shouldFetchLookups()) return;
+  final response = await get(Uri.parse(cpicLookupUrl));
+  if (response.statusCode != 200)
+    // ignore: curly_braces_in_flow_control_structures
+    throw Exception('Error while loading lookups');
+
+  final json = jsonDecode(response.body) as List<dynamic>;
+  final lookups =
+      // ignore: unnecessary_lambdas
+      json.map((el) => CpicLookup.fromJson(el)).filterValidLookups();
```

Comment:
```suggestion
  final lookups =json.map(
      (el) => CpicLookup.fromJson(el),
  ).filterValidLookups();
```
There gotta be some way to fix the linter maybe ask Benjamin
Formatting again so nothing messes with prettier

Reply:
fixed by adapting signature of `fromJson` in `CpicLookup` class in order to allow tear-off